File: TEMA3/typescript/bubblesort.ts
function bubblesort(lista: number[], ascendente: boolean){
const cantidad = lista.length;
    for(let i=0;i<cantidad-1;i++) {
        for(let j=0;j<cantidad-i-1;j++) {
         if ((ascendente && lista[j] > lista[j + 1]) ||
         (!ascendente && lista[j] < lista[j + 1]))
         //Esto no funciona se tiene que usar desestructuración
         //lista[j],lista[j+1] = lista[j+1],lista[j]
         [lista[j], lista[j + 1]] = [lista[j + 1], lista[j]];
        }
    }
    return lista
}
const list = [7, 12, 9, 11, 3]

console.log(bubblesort(list, true))
console.log(bubblesort(list,false))

File: TEMA3/python/bubblesort.py
def bubblesort(lista, ascendente=True):
    cantidad = len(lista)
    for i in range(cantidad-1):
        for j in range(cantidad-i-1):
            #aux = 0
                if (ascendente and lista[j] > lista[j + 1]) or \
                (not ascendente and lista[j] < lista[j + 1]):
                #Sin manejo de auxiliares y manejo de respectivos#
                    lista[j],lista[j+1] = lista[j+1],lista[j]
                #Con manejo de auxiliares#
                    # aux = lista[j]
                    # lista[j] = lista[j+1]
                    # lista[j+1] = aux
    return lista

list = [7, 12, 9, 11, 3]

print(bubblesort(list, True))
print(bubblesort(list,False))

